bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -0,0 +1,1331 @@
+                    updateState(testchannel, RewindFastforwardType.FASTFORWARD);
+                    break;
+            }
+            testchannel = new ChannelUID(getThing().getUID(), "playerControls", "channelArtist");
+            if (isLinked(testchannel)) {
+                updateState(testchannel, StringType.valueOf(artistState));
+            }
+            testchannel = new ChannelUID(getThing().getUID(), "playerControls", "channelTrack");
+            if (isLinked(testchannel)) {
+                updateState(testchannel, StringType.valueOf(trackState));
+            }
+            testchannel = new ChannelUID(getThing().getUID(), "playerControls", "channelAlbum");
+            if (isLinked(testchannel)) {
+                updateState(testchannel, StringType.valueOf(albumState));
+            }
+            testchannel = new ChannelUID(getThing().getUID(), "playerControls", "channelAlbumArt");
+            if (isLinked(testchannel)) {
+                if (!albumArtUrlState.equals("")) {
+                    albumArtUrlState = "http://" + this.host + albumArtUrlState;
+                }
+                updateState(testchannel, StringType.valueOf(albumArtUrlState));
+            }
+            testchannel = new ChannelUID(getThing().getUID(), "playerControls", "channelRepeat");
+            if (isLinked(testchannel)) {
+                updateState(testchannel, StringType.valueOf(repeatState));
+            }
+            testchannel = new ChannelUID(getThing().getUID(), "playerControls", "channelShuffle");
+            if (isLinked(testchannel)) {
+                updateState(testchannel, StringType.valueOf(shuffleState));
+            }
+        }
+    }
+
+    private String getResponseCode(String json) {
+        JsonElement jsonTree = parser.parse(json);
+        JsonObject jsonObject = jsonTree.getAsJsonObject();
+        return jsonObject.get("response_code").getAsString();
+    }
+
+    private String getLastInput() {
+        String text = "";
+        tmpString = getRecentInfo();
+        responseCode = getResponseCode(tmpString);
+        if (responseCode.equals("0")) {
+            JsonElement jsonTree = parser.parse(tmpString);
+            JsonObject jsonObject = jsonTree.getAsJsonObject();
+            JsonArray recentsArray = jsonObject.getAsJsonArray("recent_info");
+            for (JsonElement re : recentsArray) {
+                JsonObject recentObject = re.getAsJsonObject();
+                text = recentObject.get("text").getAsString();
+                break;
+            }
+        }
+        return text;
+    }
+
+    private void fetchOtherDevicesViaBridge() {
+        Bridge bridge = getBridge();
+        String host = "";
+        String label = "";
+        int zonesPerHost = 1;
+        List<StateOption> options = new ArrayList<>();
+        for (Thing thing : bridge.getThings()) {
+            label = thing.getLabel();
+            host = thing.getConfiguration().get("configHost").toString();
```

Comment:
`get("configHost")` can return null if the user forgets to specify it in textual configuration.

Reply:
Done, will be available with next push